### DISCUSSION TITLE
NSAttributedString property occurs crash

### DISCUSSION THREAD

**Discussion by artem-krachulov-paysera** [pointfreeco/swift-composable-architecture#2317] [Q&A]
### Description

In our project we have to use the bindable `NSAttributedString` property:
```
struct State: Equatable {
    
    /// Text bound with text area
    @BindingState public var text: String

     /// Attributed text bound with text area
    @BindingState public var attributedString: NSAttributedString?
    ...
}

```

I've added it. Two or three times I could navigate to the specific screen where this state was. Then crash appears:

![Screenshot 2023-07-21 at 5 16 44 PM](https://github.com/pointfreeco/swift-composable-architecture/assets/7997760/6f7b0bd7-f7df-4c7b-8030-711663f3c611)

When I remove `@BindingState` property wrapper I have another crash:

![Screenshot 2023-07-21 at 5 22 41 PM](https://github.com/pointfreeco/swift-composable-architecture/assets/7997760/677448fd-ff65-42ef-9991-90044552dd67)

8 of 10 times it crashes. So the crash is not 100% permanent. Also, I couldn't reproduce it in the clear project, this is very strange.

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### The Composable Architecture version information

0.55.0

### Destination operating system

16.5.1

### Xcode version information

Version 14.3.1 (14E300c)

### Swift Compiler version information

```shell
swift-driver version: 1.75.2 Apple Swift version 5.8.1 (swiftlang-5.8.0.124.5 clang-1403.0.22.11.100)
Target: x86_64-apple-macosx13.0
```

**Comment by stephencelis** [comment]
@artemkrachulov Without a project to work with I'm not sure there's much we can do here, so I'm going to convert to a discussion till it can be determined that this is a library bug.

I will note that I notice a couple strange things in the code above:

1. Why are you making `NSAttributedString` a `@BindingState`? I'm not aware of a SwiftUI control that takes a binding to an `NSAttributedString`.
2. I'm not sure `NSAttributedString` is safe to hold in app state in general. TCA state is meant to hold value types, but `NSAttributedString` is a class.

Does the crash go away entirely if you remove `NSAttributedString` from state and use a plain `String` (or `AttributedString`) instead?

**Reply by artem-krachulov-paysera** [reply]
@stephencelis thank you for your reply. Unfortunately, we use `UIViewRepresentable` with `UITextArea`, and now the project required `NSAttributedString` because we need advanced text view editing functionality.
